File: 351003/Guzaev/discussion/repositories/comment_repository.py
from cassandra.query import SimpleStatement
from database import session
from models.comment import Comment
import uuid

_counter = 1

class CommentRepository:
    def create(self, comment: Comment) -> Comment:
        global _counter
        comment.id = _counter
        _counter += 1
        session.execute(
            "INSERT INTO tbl_comment (tweet_id, country, id, content) VALUES (%s, %s, %s, %s)",
            (comment.tweet_id, comment.country, comment.id, comment.content)
        )
        return comment

    def get_all(self):
        rows = session.execute("SELECT * FROM tbl_comment")
        return [Comment(id=r.id, tweet_id=r.tweet_id, country=r.country, content=r.content) for r in rows]

    def get_by_id(self, comment_id):
        try:
            cid = int(comment_id)
        except (ValueError, TypeError):
            return None
        rows = session.execute(
            "SELECT * FROM tbl_comment WHERE id=%s ALLOW FILTERING", (cid,)
        )
        for row in rows:
            return Comment(id=row.id, tweet_id=row.tweet_id,
                           country=row.country, content=row.content)
        return None

    def delete(self, comment_id) -> bool:
        try:
            cid = int(comment_id)
        except (ValueError, TypeError):
            return False
        existing = self.get_by_id(cid)
        if not existing:
            return False
        session.execute(
            "DELETE FROM tbl_comment WHERE tweet_id=%s AND id=%s",
            (existing.tweet_id, existing.id)
        )
        return True

    def update(self, comment: Comment) -> Comment:
        session.execute(
            "INSERT INTO tbl_comment (tweet_id, country, id, content) VALUES (%s, %s, %s, %s)",
            (comment.tweet_id, comment.country, comment.id, comment.content)
        )
        return comment

File: 351002/Khmel/Task320/Repositories/CommentRepository.cs
public class CommentRepository : IRepository<Comment>
{
    private readonly AppDbContext _context;

    public CommentRepository(AppDbContext context)
    {
        _context = context;
    }

    public Comment? GetById(long id) => _context.Comments.Find(id);

    public IEnumerable<Comment> GetAll() => _context.Comments.ToList();

    public Comment Create(Comment entity)
    {
        _context.Comments.Add(entity);
        _context.SaveChanges();
        return entity;
    }

    public Comment Update(Comment entity)
    {
        var existing = _context.Comments.Find(entity.Id);
        if (existing == null)
            throw new KeyNotFoundException($"Comment с id={entity.Id} не найден");

        existing.Content = entity.Content;
        existing.StoryId = entity.StoryId;
        _context.SaveChanges();
        return existing;
    }

    public bool Delete(long id)
    {
        var entity = _context.Comments.Find(id);
        if (entity == null) return false;

        _context.Comments.Remove(entity);
        _context.SaveChanges();
        return true;
    }
}